core/arch/arm/kernel/boot.c

```diff
@@ -1310,8 +1329,16 @@ void __weak boot_init_primary_late(unsigned long fdt)
 		IMSG("Initializing virtualization support");
 		core_mmu_init_virtualization();
 	} else {
+		/* use runtime init calls to update DTB */
 		init_tee_runtime();
 	}
+
+#ifdef CFG_OVERLAY_ADDR
+	release_external_dt();
+	dt->is_overlay = 1;
+	init_external_dt(CFG_OVERLAY_ADDR);
+	update_external_dt();
```

Comment:
complete the sequence with `release_external_dt()`.

Reply:
this is taken care off by the boot final stuff - I think it is good as is.,

Reply:
That's right.